api/src/main/java/org/openmrs/validator/ObsValidator.java

```diff
@@ -223,6 +224,10 @@ else if (!obs.hasGroupMembers()) {
 				errors.rejectValue("concept", "must have a datatype");
 			}
 		}
+
+		if ( obs.getPerson() != null && obs.getEncounter() != null && obs.getEncounter().getPatient() != null && !Objects.equals( obs.getPerson(), obs.getEncounter().getPatient() )) {
```

Comment:
please format the code see https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE
you add to many spaces in your code look at the code above and below your changes it looks like

`errors.rejectValue("concept", "must have a datatype");`